rotkehlchen/history/price.py

```diff
@@ -220,10 +222,16 @@ def query_historical_price(
             except (
                 PriceQueryUnsupportedAsset,
                 NoPriceForGivenTimestamp,
-                RemoteError,
                 UnknownAsset,
             ):
                 continue
+            except RemoteError as e:
+                # Raise the flag if any of the services was rate limited
+                rate_limited = (
+                    e.error_code == HTTPStatus.TOO_MANY_REQUESTS and
+                    rate_limited is False
+                )
```

Comment:
Why check if `rate_limited` is already False? If any error code is 429 we got rate limited.

```suggestion
                rate_limited = e.error_code == HTTPStatus.TOO_MANY_REQUESTS
```

Reply:
Because say you get rate limited by coingecko, then you try cc and you get a response from their API but it is that you don't have a price. Then you would have changed `rate_limited` to false since it doesn't hold that `e.error_code == HTTPStatus.TOO_MANY_REQUESTS`. That was the two oracles you were requesting and now you hace

1. Coingecko rate limited
2. CC no price

but the status of rate_limited is False at this point

Reply:
Ah I see.

But then do it the other way. Thanks to short-circuiting you can skip the error code check

```suggestion
                rate_limited = (
                    rate_limited is True or
                    e.error_code == HTTPStatus.TOO_MANY_REQUESTS
                )
```